tests/FSharp.Stats.Tests/Integration.fs

```diff
@@ -0,0 +1,197 @@
+﻿module IntegrationTests
+open Expecto
+open System
+open FSharp.Stats
+
+open System.IO
+
+open FSharp.Stats.Integration
+
+[<Tests>]
+let numericalIntegrationTests =
+
+    /// f(x) = x^3
+    let f1 x = x * x * x
+    /// f(x) = 1/x
+    let f2 x = 1. / x
+
+    let observations1 = [|0. .. 0.0001 ..  1.|] |> Array.map (fun x -> x, f1 x)
+    let observations2 = [|1. .. 0.001 .. 100.|] |> Array.map (fun x -> x, f2 x)
+
+    //
+    let fNaN x = nan 
+    let fPosInf x = infinity 
+    let fNegInf x = -infinity
+
+    let observationsNaN = [|0. .. 0.0001 ..  1.|] |> Array.map (fun x -> x, fNaN x)
+    let observationsPosInf = [|0. .. 0.0001 ..  1.|] |> Array.map (fun x -> x, fPosInf x)
+    let observationsNegInf = [|0. .. 0.0001 ..  1.|] |> Array.map (fun x -> x, fNegInf x)
+
+    testList "NumericalIntegration" [ 
+        testList "function integration" [
+            testCase "LeftEndpoint x^3" (fun _ ->
+                let actual = f1 |> NumericalIntegration.integrateFunction(LeftEndpoint, 0., 1., 10000)
+                //exact result is 0.25
+                let expected = 0.25
+                Expect.floatClose Accuracy.low actual expected "LeftEndpoint did not return the correct result"
+            )
+            testCase "RightEndpoint x^3" (fun _ ->
+                let actual = f1 |> NumericalIntegration.integrateFunction(RightEndpoint, 0., 1., 10000)
+                //exact result is 0.25
+                let expected = 0.25
+                Expect.floatClose Accuracy.low actual expected "LeftEndpoint did not return the correct result"
+            )
+            testCase "Midpoint x^3" (fun _ ->
```

Comment:
see comment in Integration.fs
While your estimation is correct for x^3 it won't be correct for -x^3+2x^2 in the interval of [1,2]

![image](https://user-images.githubusercontent.com/28917670/162170800-7e1ca1a7-a5a7-4cf1-92ae-d4fccc654272.png)

Reply:
see above